core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java

```diff
@@ -424,9 +408,87 @@ public boolean delete(PageId pageId) {
     }
   }
 
+  @Override
+  public State state() {
+    return mState.get();
+  }
+
+  /**
+   * Restores a page store a the configured location, updating meta store.
+   */
+  private void restoreOrInit(PageStoreOptions options) {
+    Preconditions.checkState(mState.get() == READ_ONLY);
```

Comment:
One concern here is if multiple threads can go into `restoreOrInit`.

Reply:
synchronized